### ISSUE TITLE
retro and superperiods

### ISSUE THREAD

**Issue by iantaylor-NOAA** [nmfs-stock-synthesis/stock-synthesis#265]
Vlab forums message from Alessandro Orio indicates issue with superperiods in retrospectives
https://vlab.noaa.gov/web/stock-synthesis/public-forums/-/message_boards/view_message/20972432

Rick, I suspect that this one will need you to tackle as I've never looked under the hood at the super-period logic before. Please let us know if there's ways that others of us can help.

**Comment by Rick-Methot-NOAA** [comment]
I will look, but fear a generalized fix may exceed a reasonable benefit/effort.

**Comment by iantaylor-NOAA** [comment]
Thinking about this more, it occurs to me that it might not even make sense to attempt a generalized fix, regardless of the time commitment, because there's no way for the software to understand the nuances of the super-period, in terms of what data would be grouped as a super-period if fewer years of data were available.

Requiring the rare user of super-periods to manually modify 5 separate data files for a 5-year retro doesn't seem like a terrible burden considering that by the point the retros are being conducted, the data file is typically very stable.

Maybe it would be best to just augment warning such that if retro year != 0 some additional message is displayed about the need for manual edits to the super-periods.

**Comment by Rick-Methot-NOAA** [comment]
I agree.  Much higher priority to make retro play nice with time-varying parameters.

**Comment by Rick-Methot-NOAA** [comment]
new VLAB forum post said that superperiods and retro worked in 3.24, so I will investigate this issue for 3.30.19.  Example files are attached to the [VLAB post.](https://vlab.noaa.gov/web/stock-synthesis/public-forums/-/message_boards/message/20996032)

**Comment by Rick-Methot-NOAA** [comment]
The model stops because of a new fatal warning that was added to detect erroneous situations with multiple real observations in a superperiod.

With the retroyr, there are zero real obs in the terminal super-years.  However the warning condition was set to <>1, rather than >1, so the model stops.

So, this should be a rather easy fix in the preliminary calcs section of the code.

**Comment by Rick-Methot-NOAA** [comment]
Let's leave this issue open for potential future action.   A partial fix was implemented for lencomp superperiods such that a fatal exit would not be triggered, but any superperiod than spans retroyr will not be implemented as intended and the user will get no warning.
